src/FakeItEasy/Core/TypeCatalogue.cs

```diff
@@ -91,7 +91,7 @@ private static IEnumerable<Assembly> GetAllAssemblies(IEnumerable<string> extraA
                     // Exclude the ReflectionOnly assemblies because we may fully load them later.
                     .Where(a => !a.ReflectionOnly)
 #endif
-                    .Where(a => !a.IsDynamic)
+                    .Where(a => !a.IsDynamic && !a.Name().Equals("DynamicProxyGenAssembly2", StringComparison.OrdinalIgnoreCase))

```

Comment:
Odd. I'm struggling with this issue. On the one hand, I believe that you've seen this problem. I just struggle to  understand how. And without understanding how, I have less confidence in the fix.

First, it's rare to reference `DynamicProxyGenAssembly2` before `TypeCatalogue.Load` is called.
Have you seen the error in the wild, or only while running the integration tests?
Are you able to reproduce it at will? How?
Do you know where the error was triggered? Is it the same as in #1092, where you see problems loading a particular type?

I've hacked up the code to throw when it sees `DynamicProxyGenAssembly2`, and to record its dynamicness, and it invariably says that it's dynamic.

Help me believe in the fix?